index.html

```diff
@@ -323,6 +323,140 @@
+            (for instance, a URL), then return a <a>Promise</a> rejected with a <a>SecurityError</a>.
+          </li>
+          <li>
+            If there is already an unsettled <a>Promise</a> from a previous
+            call to <code>connect</code> for the same browsing context, return
+            a <a>Promise</a> rejected with an <a>OperationError</a> exception and
+            abort all remaining steps.
+          </li>
+          <li>
+            Let <var>promise</var> be a new <a>Promise</a>.
+          </li>
+          <li>
+            Return <var>promise</var> and continue running these steps
+            asynchronously.
+          </li>
+          <li>
+            If the <a>user agent</a> is not <a data-lt="monitor the list of available remote playback devices">
+            monitoring the list of available remote playback devices</a>, run the steps to
+            <a>monitor the list of available remote playback devices</a> in parallel.
+          </li>
+          <li>
+            Request user permission for the use of a <a>remote playback device</a>
+            and selection of one remote playback device.
+          </li>
+          <li>
+            If either of the following is true:
+            <ol>
+              <li>
+                The <a>list of available remote playback devices</a> is empty
+                and will remain so before the request for user permission is
+                completed.
+              </li>
+              <li>
+                No member in the <a>list of available remote playback device</a> is a
+                <a>compatible remote playback device</a> for
+                <var>remotePlaybackSource</var>.
+              </li>
+            </ol>
+            Then run the following steps:
+            <ol>
+              <li>
+                <a>Reject</a> <var>promise</var> with a <a>NotFoundError</a> exception.
+              </li>
+              <li>
+                Abort all remaining steps.
+              </li>
+            </ol>
+          </li>
+          <li>
+            If the user <em>denied permission</em> to use a device, resolve
+            <var>promise</var> with <code>false</code>, and abort
+            all remaining steps.
+          </li>
+          <li>
+            Otherwise, the user <em>granted permission</em> to use a device;
+            let <var>device</var> be that device and <var>userAgent</var>
+            the user agent connected to <var>device</var>.
+          </li>
+          <li>
+            Set the <a for="RemotePlayback">state</a> of the <var>remote</var> object to
+            <a for="RemotePlabyackState">connecting</a>.
+          </li>
+          <li>
+            <a>Resolve</a> <var>promise</var> with <code>true</code>.
+          </li>
+          <li>
+            <a>Queue a task</a> to <a>fire</a> a <a>trusted event</a> with
+            the name <a for="PresentationRequest">statechange</a>,
```

Comment:
I think you can use `simple event`: https://www.w3.org/TR/html5/webappapis.html#fire-a-simple-event